feat: Organization VPCs and org VPC peering (#579)
aiven/aiven-docs: docs/platform/howto/vnet-peering-azure.md

  permissions in Aiven
- Azure account with at least the application administrator role
- [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/?view=azure-cli-latest) and,
  optionally, the [Microsoft Azure portal](https://portal.azure.com/#home)
- [Aiven CLI](/docs/tools/cli) installed

## Set up permissions in Azure

### Azure app object permissions

1. Log in with an Azure admin account using the Azure CLI:

   ```bash
   az account clear
   az login
   ```

   This should open a window in your browser prompting to choose an Azure
   account to log in with.

   :::tip
   If you manage multiple Azure subscriptions, also configure the Azure CLI
   to default to the correct subscription for the subsequent commands. This
   is not needed if there's only one subscription:

   ```bash
   az account set --subscription SUBSCRIPTION_NAME_OR_ID
   ```

   :::

1. Create an application object in your AD tenant using the Azure CLI:

   ```bash
   az ad app create                         \
     --display-name "NAME_OF_YOUR_CHOICE"   \
     --sign-in-audience AzureADMultipleOrgs \
     --key-type Password
   ```

>    This creates an entity to your AD that can be used to log into multiple

Reviewer: ```suggestion
   This creates an application object in Azure AD that can be used to log into multiple
```

   AD tenants ( `--sign-in-audience AzureADMultipleOrgs` ), but only the
   home tenant (the tenant the app was created in) has the credentials to
   authenticate the app.

   :::note
   Save the `appId` field from the output. It will be referred to as `$user_app_id`.
   :::

1. Create a service principal for your app object to the Azure subscription that the VNet
   to be peered is located in:

   ```bash
   az ad sp create --id $user_app_id
   ```

>    This creates a service principal to your subscription that may be given

Reviewer: ```suggestion
   This creates a service principal in your subscription, which can be 
   assigned permissions to peer your VNet.
```

       "Actions": ["Microsoft.Network/virtualNetworks/peer/action"],
       "AssignableScopes": ["/subscriptions/'$user_subscription_id'"]
     }'
   ```

   `AssignableScopes` includes your Azure subscription ID to restrict scopes that a role
   assignment can use.

   :::note
   Save the `id` field from the output. It will be referred to as `$aiven_role_id`.
   :::

1. Assign the custom role to the Aiven service principal.

   To give the Aiven application object's service principal permissions to peer with your
   VNet, assign the created role to the Aiven service principal with the scope of your
   VNet:

   ```bash
   az role assignment create           \
     --role $aiven_role_id             \
     --assignee-object-id $aiven_sp_id \
     --scope $user_vnet_id
   ```

1. Find your AD tenant ID:

   - In the [Azure portal](https://portal.azure.com/#home): **Settings** >
     **Directories + subscriptions** > **Directories** > **Directory ID**
   - Using the Azure CLI:

     ```bash
     az account list
     ```

   :::note
   Save the `tenantId` field from the output. It will be referred to as `$user_tenant_id`.
   :::

## Create the peering in Aiven

> By creating a peering from the Aiven project VPC to the VNet in your Azure subscription,

Reviewer: ```suggestion
By creating a peering connection from the Aiven project VPC to the VNet in your Azure subscription,
```

you also create a service principal for the application object
(`--peer-azure-app-id $user_app_id`) and grant it the permission to peer with the Aiven
project VPC.

The Aiven application object authenticates with your Azure tenant to grant it access to
[the service principal of the Aiven application object](/docs/platform/howto/vnet-peering-azure#aiven-app-object-permissions)
(`--peer-azure-tenant-id $user_tenant_id`).

> `$aiven_project_vpc_id` can be found in the [Aiven Console](https://console.aiven.io/) or
> using the `avn vpc list` command.

Reviewer: ```suggestion
Find `$aiven_project_vpc_id` in the [Aiven Console](https://console.aiven.io/) or by running the `avn vpc list` command.
```
